spec/post_spec.rb

```diff
@@ -108,6 +110,32 @@
       capture_stdout { described_class.process(args) }
       expect(path).to exist
     end
+
+    context 'auto_open editor is set' do
+      let(:posts_dir) { Pathname.new source_dir("_posts") }
+      let(:config_data) { %(
+jekyll_compose:
+  auto_open: true
+)}
+
+      context 'env variable EDITOR is set up' do
+        before { ENV['EDITOR'] = 'vim' }
+
+        it 'opens post in default editor' do
+          allow(Jekyll::Compose::FileEditor).to receive(:run_editor).with('vim', path.to_s)
```

Comment:
I think this should be `expect` instead of `allow`, otherwise it would also pass if `Jekyll::Compose::FileEditor` never receives `#run_editor`.